### PR TITLE
Button custom content additions

### PR DESCRIPTION
**Pull request by philschanely**
## Description

This PR adds a few more features to the recent custom content in buttons work:

- Adds ability to add a class on the custom content container.
- Adds corresponding features in React Buttons

## Testing in `sage-lib`

- See Button page and the first row of buttons where the custom content example on the end now shows a custom class with some flexbox styling.
- See Storybook > Button > Custom Content for another similar example.

## Testing in `kajabi-products`
(LOW) Updates to buttons with custom content. No impact on existing buttons expected.

## Related
#586

### CODE REVIEW

File: docs/app/views/examples/components/button/_props.html.erb [hunk 1]
```diff
@@ -10,6 +10,17 @@
   <td><%= md('Hash') %></td>
   <td><%= md('`nil`') %></td>
 </tr>
+<tr>
+  <td><%= md('`custom_content_class`') %></td>
+  <td>
+    <%= md('
+      Passes a custom CSS class for when custom HTML content is provided as content in the component.
+      We recommend this be used primarily for controlling the layout of the custom content and to avoid any color modifications that would overrides system styles for buttons.
```
Comment (pixelflips) [inline docs/app/views/examples/components/button/_props.html.erb:18]: 😍

**Review by pixelflips — APPROVED** [review]
LGTM! 👍🏼

File: docs/lib/sage-frontend/stylesheets/docs/_example.scss [hunk 2]
```diff
@@ -174,3 +174,8 @@ $-example-code-preview-button-bg: rgba(sage-color(primary, 500), 0.75);
     }
   }
 }
+
+.demo-custom-class {
+  display: flex;
+  gap: 8px;
```
Comment (QuintonJason) [inline docs/lib/sage-frontend/stylesheets/docs/_example.scss:180]: can this reference the `sage-spacing` var?

Comment (philschanely) [reply docs/lib/sage-frontend/stylesheets/docs/_example.scss:180]: Sure thing. Thanks!

**Pull request merged**